cmake/Platform/Windows/Packaging/Tests/test_installer.py

```diff
@@ -0,0 +1,265 @@
+
+    # the install root is created
+    assert context.install_root.is_dir(), f"Invalid install root {context.install_root}"
+
+@pytest.mark.parametrize(
+    "filename", [
+        pytest.param('Editor.exe'),
+        pytest.param('AssetProcessor.exe'),
+        pytest.param('MaterialEditor.exe'),
+        pytest.param('o3de.exe'),
+    ]
+)
+def test_binaries_exist(test_installer_fixture, context, filename):
+    """Installer key binaries exist"""
+    assert (context.engine_bin_path / filename).is_file(), f"{filename} not found in {context.engine_bin_path}"
+
+@pytest.fixture(scope="session")
+def test_o3de_registers_engine_fixture(test_installer_fixture, context):
+    """Engine is registered when o3de.exe is run"""
+    try:
+        result = context.run([str(context.engine_bin_path / 'o3de.exe')], timeout=7)
+
+        # o3de.exe should not close with an error code 
+        assert result.returncode == 0, f"o3de.exe failed with exit code {result.returncode}"
+    except TimeoutExpired as e:
+        # we expect to close the app ourselves
+        pass
+
+    # the engine is registered
+    engine_json_path = context.install_root / 'engine.json'
+    with engine_json_path.open('r') as f:
+        engine_json_data = json.load(f)
+
+    engine_name = engine_json_data['engine_name']
+
+    manifest_path = Path(os.path.expanduser("~")).resolve() / '.o3de' / 'o3de_manifest.json'
+    with manifest_path.open('r') as f:
+        manifest_json_data = json.load(f)
+    
+    engine_path = Path(context.install_root).as_posix()
+    assert engine_path in manifest_json_data['engines'] , f"Engine path {engine_path} not found in o3de_manifest.json"
+    assert engine_name in manifest_json_data['engines_path'], f"{engine_path} not found in manifest engines_path"
+    assert manifest_json_data['engines_path'][engine_name] == engine_path, f"Engines path has invalid entry for {engine_name} expected {engine_path}" 
+
+@pytest.fixture(scope="session")
+def test_create_project_fixture(test_o3de_registers_engine_fixture, context):
+    """o3de CLI creates a project"""
+    o3de_path = context.install_root / 'scripts' / 'o3de.bat'
+    result = context.run([str(o3de_path),'create-project','--project-path', str(context.project_path)])
+    assert result.returncode == 0, f"o3de.bat failed to create a project with exit code {result.returncode}"
+
+    project_json_path = Path(context.project_path) / 'project.json'
+    assert project_json_path.is_file(), f"No project.json found at {project_json_path}"
+
+@pytest.fixture(scope="session")
+def test_compile_project_fixture(test_create_project_fixture, context):
+    """Project can be configured and compiled"""
+    project_name = Path(context.project_path).name
+    launcher_target = f"{project_name}.GameLauncher"
+
+    # configure
+    result = context.run([str(context.cmake_path),'-B', str(context.project_build_path), '-S', '.', '-G', 'Visual Studio 16 2019'], cwd=context.project_path)
```

Comment:
Do we also want to test compiling on 2022? Also do we want to just change this test to the newest version every few years, or is there a better way to detect the version of VS we should use?

Reply:
At some point we will definitely want to update this for 2022 support - but that can be a future PR